mantidproject/mantid
# Convert a Python dict to a C++ PropertyManager for algorithms

Introduces a handler class in the Python layer than can convert from a Python `dict` to a C++ `PropertyManagerProperty` for more convenient passing of dictionary types for an algorithm interface.

**To test:**

The majority of the changes came from moving `PropertyManagerDataService` from `API` to `Kernel`.

Examine the new `PropertyManagerPropertyTest` changes to see how a Python algorithm would look with this new type and play with the interface to it.

Fixes #16029

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] If there are changes in the release notes then do they describe the changes appropriately? ->N/A
##### Functional Tests
- [ ] Has the relevant documentation been added/updated? ->N/A
- [ ] Is user-facing documentation written in a user-friendly manner? ->N/A
- [ ] Has developer documentation been updated if required? ->N/A
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
